sql/2020/20_Caching/cache_ttl_and_content_age_diff.sql

```diff
@@ -0,0 +1,34 @@
+#standardSQL
+# Difference between Cache TTL and the content age
+CREATE TEMPORARY FUNCTION toTimestamp(date_string STRING)
+RETURNS INT64 LANGUAGE js AS '''
+  try {
+    var timestamp = Math.round(new Date(date_string).getTime() / 1000);
+    return isNaN(timestamp) ? -1 : timestamp;
+  } catch (e) {
+    return -1;
+  }
+''';
+
+SELECT
+  client,
+  percentile,
+  APPROX_QUANTILES(diff_in_days, 1000)[OFFSET(percentile * 10)] AS diff_in_days
+FROM
+  (
+    SELECT
+      client,
+      ROUND((expAge - (startedDateTime - toTimestamp(resp_last_modified))) / 86400, 2)  AS diff_in_days
+    FROM
+      `httparchive.almanac.requests`
+    WHERE
+      date = '2020-08-01' AND
+      resp_last_modified <> "" AND expAge > 0
+  ),
+  UNNEST([10, 20, 30, 40, 50, 60, 70, 80, 90]) AS percentile

```

Comment:
(throughout) these are the percentiles used most frequently by other chapters
```suggestion
  UNNEST([10, 25, 50, 75, 90]) AS percentile
```

Reply:
percentiles in all queries now have only 10, 25, 50, 75, and 90.